routing/dht/routing.go

```diff
@@ -153,7 +168,7 @@ func (dht *IpfsDHT) Provide(ctx context.Context, key u.Key) error {
 // FindProviders searches until the context expires.
 func (dht *IpfsDHT) FindProviders(ctx context.Context, key u.Key) ([]peer.PeerInfo, error) {
 	var providers []peer.PeerInfo
-	for p := range dht.FindProvidersAsync(ctx, key, math.MaxInt32) {
+	for p := range dht.FindProvidersAsync(ctx, key, KValue) {
```

Comment:
Because its wasteful, we are getting to the point where there might actually be more than K providers of a value, and as it stands now, we will send out requests to each and every single one of them, which will make them send us blocks, lots of wasted bandwidth.